### PR TITLE
DEV-1952: Remove pyrsistent from install deps.

### PR DESCRIPTION
**Pull request by wrkhenddher**
Also:
- Upgrade tox.ini (allowlist_externals vs whitelist_externals and remove reqs.txt from install. dev-reqs has everything)
- Add --add-opens java.base/java.util=ALL-UNNAMED to Java build command for swagger_client (Mac M2 installation notes).
- Add extras_require for dev deps to setup.py.
- Remove dev-requirements.in. When regenerating dev-reqs.txt, --pip-args='-c requirements.txt' is needed to specify *constraints*.

### CODE REVIEW

**Comment by qiaouchicago** [comment]
Please remove dev-requirements.in and regenerate dev-requirements.txt from setup.py

File: dev-requirements.txt [hunk 1]
```diff
@@ -2,114 +2,139 @@
...
-    #   flask
+    # via flask
 coverage[toml]==6.2
     # via pytest-cov
-dataclasses==0.8 ; python_version < "3.7"
-    # via
-    #   -c requirements.txt
-    #   werkzeug
+dataclasses==0.8
+    # via werkzeug
```
Comment (wrkhenddher) [inline dev-requirements.txt:28]: dataclasses is being added by werkzug **because** python 3.6 is being used still.
👎🏻

**Comment by wrkhenddher** [comment]
> Please remove dev-requirements.in and regenerate dev-requirements.txt from setup.py

Bad news @qiaouchicago, regenerating both reqs files caused the build to break because of dataclasses.

**Comment by wrkhenddher** [comment]
~~Bad news @qiaouchicago, regenerating both reqs files caused the build to break because of dataclasses.~~

Nm, it's fine.

I was able to regen dev-reqs.txt (reusing existing dev-reqs.txt) with reqs.txt as constraints.

**Review by qiaouchicago — APPROVED** [review]

**Review by XukunXiangUChicago — APPROVED** [review]

**Pull request merged**